### PR TITLE
Fix Safari runtime error for..of weirdness

### PR DESCRIPTION
**Pull request by luwes**
We've been using template-parts in our new video player project over at Mux. 
We really like it with the added utilities of jtml! 

Today we ran into a rather strange runtime error only in Safari and it only happens when the web inspector is not open.
Try loading this page with and without web inspector open, might need a few reloads:
https://elements-demo-nextjs-git-fork-luwes-better-errors-mux.vercel.app/MuxPlayer

1. Without web inspector open, the video player layout sometimes doesn't look good, the layout of icons and buttons are wrong.
2. If you then open the web inspector you see a type error like

```js
TypeError: e of xr is not a function. (In 'e of xr(t.textContent)', 'e of xr' is undefined)
```
This refers to this line:
https://github.com/github/template-parts/blob/main/src/template-instance.ts#L28

I don't think the code is wrong here but Safari definitely misinterprets it somehow.
Changing the for..of loop to a regular for loop fixes the issue,
I also tried to store the result of the parse() function in a variable first and use the for..of loop but that didn't fix it :(

SCR-20220325-d8k 

SCR-20220325-dye

Mac OS 12.3
Safari 15.4

### CODE REVIEW

**Comment by luwes** [comment]
mobile Safari on iOS 12 and probably other versions also seems to be affected
also seeing the console output here it shows it's throwing in native code and related to the generator, seeing `generatorResume` there. very much looks like a browser bug unfortunately.


SCR-20220325-srl

**Review by keithamus — APPROVED** [review]
That's super weird! Thanks for the fix!

File: src/template-instance.ts [hunk 1]
```diff
@@ -25,7 +25,9 @@ function* collectParts(el: DocumentFragment): Generator<TemplatePart> {
         }
       }
     } else if (node instanceof Text && node.textContent && node.textContent.includes('{{')) {
-      for (const token of parse(node.textContent)) {
+      const parsed = parse(node.textContent)
+      for (let i = 0; i < parsed.length; i += 1) {
+        const token = parsed[i]
```
Comment (dylanjha) [inline src/template-instance.ts:30]: Small comment: Do you think a code comment with a link to this PR for context would be helpful?

I'd be a little worried that someone sees this code and refactors it back to `for const ... in`, resulting in a regression.

Comment (keithamus) [reply src/template-instance.ts:30]: This sounds like a great idea! 

I'll merge this once a comment is added 😝

**Comment by dylanjha** [comment]
@keithamus I think this is ready now 🙌🏼 thanks for your help!

**Comment by dylanjha** [comment]
Will a new version be coming out soon :) we're currently pointed to a fork to get this patch, but we would love to point back upstream when this is released!

**Comment by cjpillsbury** [comment]
Hey @keithamus just checking in to see if there were any plans to do an npm release that includes this update.

**Comment by keithamus** [comment]
Oh sorry! This one slipped by :grimacing:. Should be out as `v0.5.2`

**Comment by keithamus** [comment]
Oh no 🙈. We seem to be having some publish difficulties. I'll make sure to get these resolved soon!

**Pull request merged**